.github/workflows/hls.yml

```diff
@@ -6,8 +6,53 @@ on:
 
 jobs:
 
+  build-circt:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v4
+    - name: "Install dependencies"
+      uses: ./.github/actions/InstallPackages
+      with:
+        install-llvm: true
+        install-clang: true
+        install-valgrind: false
+    - name: Set color
+      id: color-selector
+      run: echo "SELECTED_COLOR=green" >> "$GITHUB_OUTPUT"
+    - name: Get color
+      env:
+        SELECTED_COLOR: ${{ steps.color-selector.outputs.SELECTED_COLOR }}
+      run: echo "The selected color is $SELECTED_COLOR"
+    - name: "Get the commit used for building CIRCT and use it as the cache key"
+      id: get-circt-hash
+      run: |
+        echo "GIT_HASH=$(./scripts/build-circt.sh --get-commit-hash)" >> "$GITHUB_OUTPUT"
```

Comment:
The difference to the `SELECTED_COLOR=green` above is that you do not just assign a string statically here. You are trying to execute something. I suspect something silently fails there. What happens if you just do statically `"GIT_HASH=<copied-hash-key-from-script>"` and then try to run it again? If it works, then you know that something is wrong with `$(./scripts/build-circt.sh --get-commit-hash)`